src/components/BotSetting/BotSetting.tsx

```diff
@@ -1,7 +1,115 @@
+import { Typography } from '@mochi-ui/core'
+import { BotDescription } from './BotDescription'
+import { BotModel } from './BotModel'
+import { BotMessages } from './BotMessages'
+import { BotLimit } from './BotLimit'
+import { FormProvider, useForm } from 'react-hook-form'
+import { UsageLimitTypeEnum } from '~/model/usage-limit-type'
+import { BotModelEnum } from '~/model/bot-model'
+import { Button } from '@mochi-ui/core'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+import { useParams } from 'next/navigation'
+import { api } from '~/utils/api'
+
+export interface BotSettingData {
+  botId: string
+  name: string
+  description: string
+  modelId?: BotModelEnum | undefined
+  messageNoSource?: string | undefined
+  messageServerError?: string | undefined
+  usageLimitPerUser?: number | undefined
+  usageLimitPerUserType?: UsageLimitTypeEnum | undefined
+  userLimitWarningMsg?: string | undefined
+}
+
+const schema = z.object({
+  botId: z.string(),
+  name: z
+    .string()
+    .min(1, 'Required')
+    .max(50, 'Should not exceed 50 characters.'),
+  description: z.string().max(500, 'Should not exceed 500 characters.'),
```

Comment:
A bit shorter 😓 

```suggestion
  description: z.string().max(500, 'Max length is 500 characters.'),
```

Please update all other messages